src/peft/peft_model.py

```diff
@@ -258,6 +258,13 @@ def save_pretrained(
             path_initial_model_for_weight_conversion = convert_pissa_to_lora
 
         def save_mutated_as_lora(peft_config, path_initial_model_for_weight_conversion, output_state_dict, kwargs):
+            if peft_config.use_rslora and (peft_config.rank_pattern or peft_config.alpha_pattern):
```

Comment:
Can't we use this condition as soon as this permutation of config is requested and warn/error?

Reply:
Yes, this is done below and we warn the user: https://github.com/huggingface/peft/pull/1930/files#diff-13380145401d203d5935c5189dd09879f990b81aa63e8e3aaff8ce9110333f0eR354

We don't raise an error since we cannot know at that point in time if they'll use the `path_initial_model_for_weight_conversion`. If they don't, there is no problem.

Reply:
Okay that should cut it.